File: JAVA/Clothing.java
/*
    Saya Revana Faliha Salma NIM 2202869 mengerjakan Soal Latihan 2 
    dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak 
    melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

class Clothing extends Product{
    private String size;    // deklarasi variabel untuk menyimpan ukuran pakaian
    private String material;    // deklarasi variabel untuk menyimpan bahan pakaian
    private String gender;  // deklarasi variabel untuk menyimpan jenis kelamin yang ditujukan pakaian

    // konstruktor kosong
    Clothing(){

    }

    // konstruktor parameter untuk menginisialisasi objek Clothing dengan nilai yang diberikan
    public Clothing(String size, String material, String gender){
        this.size = size;   // inisialisasi ukuran pakaian
        this.material = material;   // inisialisasi bahan pakaian
        this.gender = gender;   // inisialisasi jenis kelamin yang ditujukan pakaian
    }

    // metode setter untuk mengatur ukuran pakaian
    public void setSize(String size){
        this.size = size;
    }

    // metode getter untuk mendapatkan ukuran pakaian
    public String getSize(){
        return this.size;
    }

    // metode setter untuk mengatur bahan pakaian
    public void setMaterial(String mat){
        this.material = mat;
    }

    // metode getter untuk mendapatkan bahan pakaian
    public String getMaterial(){
        return this.material;
    }

    // metode setter untuk mengatur jenis kelamin yang ditujukan pakaian
    public void setGender(String gender){
        this.gender = gender;
    }

    // metode getter untuk mendapatkan jenis kelamin yang ditujukan pakaian
    public String getGender(){
        return this.gender;
    }
}

File: PYTHON/Program/Clothing.py
# Saya Revana Faliha Salma NIM 2202869 mengerjakan Soal Latihan 2 
# dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak 
# melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

from Product import Product  # mengimpor kelas Product untuk inheritance

class Clothing(Product):  # mendefinisikan kelas Clothing yang mewarisi dari kelas Product
    __size = ""  # variabel privat untuk ukuran pakaian
    __material = ""  # variabel privat untuk bahan pakaian
    __gender = ""  # variabel privat untuk jenis kelamin yang cocok

    def __init__(self, size = "", material = "", gender = ""):
        super().__init__()  # memanggil konstruktor kelas induk (Product) dan menginisialisasi atributnya
        self.__size = size  # menginisialisasi atribut ukuran pakaian
        self.__material = material  # menginisialisasi atribut bahan pakaian
        self.__gender = gender   # menginisialisasi atribut jenis kelamin yang cocok

    # metode untuk mengatur ukuran pakaian
    def set_size(self, size):
        self.__size = size

    # metode untuk mendapatkan ukuran pakaian
    def get_size(self):
        return self.__size
    
    # metode untuk mengatur bahan pakaian
    def set_material(self, material):
        self.__material = material

    # metode untuk mendapatkan bahan pakaian
    def get_material(self):
        return self.__material
    
    # metode untuk mengatur jenis kelamin yang cocok
    def set_gender(self, gender):
        self.__gender = gender

    # metode untuk mendapatkan jenis kelamin yang cocok
    def get_gender(self):
        return self.__gender
